syntax = "proto3";

package calculator;

service Calculator {
    rpc Add (AddRequest) returns (AddResponse);
    rpc Subtract (SubtractRequest) returns (SubtractResponse);
    rpc Multiply (MultiplyRequest) returns (MultiplyResponse);
    rpc Divide (DivideRequest) returns (DivideResponse);
    rpc SquareRoot (SquareRootRequest) returns (SquareRootResponse);
}

message AddRequest {
    int32 num1 = 1;
    int32 num2 = 2;
}

message AddResponse {
    int32 result = 1;
}

message SubtractRequest {
    int32 num1 = 1;
    int32 num2 = 2;
}

message SubtractResponse {
    int32 result = 1;
}

message MultiplyRequest {
    int32 num1 = 1;
    int32 num2 = 2;
}

message MultiplyResponse {
    int32 result = 1;
}

message DivideRequest {
    int32 num1 = 1;
    int32 num2 = 2;
}

message DivideResponse {
    float result = 1;
}

message SquareRootRequest {
    float num = 1;
}

message SquareRootResponse {
    float result = 1;
}